File: src/Database/Repositories/BlacklistRepository.cs
using DEA.Database.Models;
using MongoDB.Driver;
using System.Threading.Tasks;

namespace DEA.Database.Repositories
{
    public sealed class BlacklistRepository : BaseRepository<Blacklist>
    {
        public BlacklistRepository(IMongoCollection<Blacklist> blacklists) : base(blacklists) { }

        public Task AddGuildAsync(ulong userId, ulong newGuildId)
        {
            return PushAsync(c => c.UserId == userId, "GuildIds", (decimal)newGuildId);
        }
    }
}

File: src/database/repositories/BlacklistRepository.js
const BaseRepository = require('./BaseRepository.js');
const PushUpdate = require('../updates/PushUpdate.js');
const BlacklistQuery = require('../queries/BlacklistQuery.js');
const Blacklist = require('../models/Blacklist.js');

class BlacklistRepository extends BaseRepository {
  anyBlacklist(userId) {
    return this.any(new BlacklistQuery(userId));
  }

  findBlacklist(userId) {
    return this.findOne(new BlacklistQuery(userId));
  }

  insertBlacklist(userId, username, avatarURL) {
    return this.insertOne(new Blacklist(userId, username, avatarURL));
  }

  addGuild(userId, newGuildId) {
    return this.updateOne(new BlacklistQuery(userId), new PushUpdate('guildIds', newGuildId));
  }
}

module.exports = BlacklistRepository;
